### ISSUE TITLE
Linux version?

### ISSUE THREAD

**Issue by moondeck** [Phoneboard/Phoneboard#4]
Hi, would it be possible to have a Linux version of Phoneboard? It doesn't seem to run with Wine.

Thanks!

**Comment by guidopola** [comment]
Yeah I only have to figure how to create the deb / rpm packages, I just wanted to release v1.1.0 before looking into it. 
Btw is running fine on my box with wine.

**Comment by moondeck** [comment]
When i was creating a .deb package this helped a lot https://wiki.debian.org/HowToPackageForDebian

You might wanna consider using AppImage though. I am really for applications using it since it is distro-independent and there are no dependencies, you might want to look into that: https://appimage.org/

I would be happy to help with it if you need someone to do this.

**Comment by moondeck** [comment]
It works! Thank you!
Seems like some icons are pixelated though (left "Phones" and "Tablets", rotation/zoom/others on top)

Here's the log if it helps you debugging:

`$ ./phoneboard-linux-v1.1.0-x86_64.AppImage
qt.network.ssl: QSslSocket: cannot resolve CRYPTO_num_locks
qt.network.ssl: QSslSocket: cannot resolve CRYPTO_set_id_callback
qt.network.ssl: QSslSocket: cannot resolve CRYPTO_set_locking_callback
qt.network.ssl: QSslSocket: cannot resolve ERR_free_strings
qt.network.ssl: QSslSocket: cannot resolve EVP_CIPHER_CTX_cleanup
qt.network.ssl: QSslSocket: cannot resolve EVP_CIPHER_CTX_init
qt.network.ssl: QSslSocket: cannot resolve sk_new_null
qt.network.ssl: QSslSocket: cannot resolve sk_push
qt.network.ssl: QSslSocket: cannot resolve sk_free
qt.network.ssl: QSslSocket: cannot resolve sk_num
qt.network.ssl: QSslSocket: cannot resolve sk_pop_free
qt.network.ssl: QSslSocket: cannot resolve sk_value
qt.network.ssl: QSslSocket: cannot resolve SSL_library_init
qt.network.ssl: QSslSocket: cannot resolve SSL_load_error_strings
qt.network.ssl: QSslSocket: cannot resolve SSL_get_ex_new_index
qt.network.ssl: QSslSocket: cannot resolve SSLv3_client_method
qt.network.ssl: QSslSocket: cannot resolve SSLv23_client_method
qt.network.ssl: QSslSocket: cannot resolve SSLv3_server_method
qt.network.ssl: QSslSocket: cannot resolve SSLv23_server_method
qt.network.ssl: QSslSocket: cannot resolve X509_STORE_CTX_get_chain
qt.network.ssl: QSslSocket: cannot resolve OPENSSL_add_all_algorithms_noconf
qt.network.ssl: QSslSocket: cannot resolve OPENSSL_add_all_algorithms_conf
qt.network.ssl: QSslSocket: cannot resolve SSLeay
qt.network.ssl: Incompatible version of OpenSSL
qt.network.ssl: QSslSocket: cannot call unresolved function SSLv23_client_method
qt.network.ssl: QSslSocket: cannot call unresolved function SSL_CTX_new
qt.network.ssl: QSslSocket: cannot call unresolved function SSL_library_init
qt.network.ssl: QSslSocket: cannot call unresolved function ERR_get_error
qt.network.ssl: QSslSocket: cannot call unresolved function ERR_get_error
`

**Comment by moondeck** [comment]
Actually, just tried opening the schematics for iP6, got a segfault!

'QFont::setPointSize: Point size <= 0 (0), must be greater than 0
QFont::setPointSize: Point size <= 0 (0), must be greater than 0
Segmentation fault'

**Comment by guidopola** [comment]
The ssl error are because qt5 uses an old openssl lib, seems like it's not embeded in the AppImage.
what are your display resolution? are a normal display or a HiDPI one?

**Comment by guidopola** [comment]
> The ssl error are because qt5 uses an old openssl lib, seems like it's not embeded in the AppImage.
> what are your display resolution? are a normal display or a HiDPI one?

>> Actually, just tried opening the schematics for iP6, got a segfault!
>> 
>> 'QFont::setPointSize: Point size <= 0 (0), must be greater than 0
>> QFont::setPointSize: Point size <= 0 (0), must be greater than 0
>> Segmentation fault'

About the segfault, maybe it's the bug i'm tracking right now. If you load a board file and move the cursor inside the window sometimes it crashes.

**Comment by moondeck** [comment]
1920x1200, on a ThinkPad T61, T8100, 4 gigs of RAM. Perhaps it is worth mentioning i am running it in a dual display config now.

**Comment by guidopola** [comment]
The problem can be related with the font rendering (I'm using FontAwesome for the icons), I can replace them with SVG right now and upload a new AppImage if you have some free time to test

**Comment by moondeck** [comment]
Sure, no problem.

**Comment by moondeck** [comment]
Perhaps it could be related to my (rather lousy) hardware. It's a laptop, and it is 10 years old. I'll try and see how the fonts look on a more powerful desktop tomorrow.

**Comment by guidopola** [comment]
I can't manage to get svg icons look nice. But i have been scaling the fontawesome icons, so i'm uploading a non-scaling version to test.
Btw, I code on a 2014 laptop with 4gigs of ram with an intel pentium processor, so don't worry about old hardware ;)

**Comment by guidopola** [comment]
Looking at the screenshot the issue seems to be related to font size, can you check running the app with 
`QT_SCALE_FACTOR=1.5` env var?
`QT_SCALE_FACTOR=1.5 ./Phoneboard-test-v1.1.0-x86_64.AppImage`

**Comment by moondeck** [comment]
Works fine on 1920x1080 now, kinda big for 1920x1200, but the screen is also 21.5" vs 15.6". The icons look even worse now.

**Comment by moondeck** [comment]
Perhaps add an option for changing scaling in the settings?

**Comment by guidopola** [comment]
Good to know!
